util/minetestmapper.py

```diff
@@ -190,22 +199,126 @@ def usage():
+            nonchunky_xmax = nonchunky_xmin + this_width - 1  # inclusive rect
+            nonchunky_zmin = z
+            nonchunky_zmax = nonchunky_zmin + this_height - 1  # inclusive rect
+            print("#geometry:")
+            print("#  x:" + str(x))
+            print("#  z:" + str(z))
+            print("#  width:" + str(this_width))
+            print("#  height:" + str(this_height))
+            print("region:")
+            print("  xmin:" + str(nonchunky_xmin))
+            print("  xmax:" + str(nonchunky_xmax))
+            print("  zmin:" + str(nonchunky_zmin))
+            print("  zmax:" + str(nonchunky_zmax))
+        else:
+            print("ERROR: Missing coordinates in '" + geometry_string +
+                  "' for geometry (must be in the form: x:z+width+height)")
+            usage()
+            sys.exit(2)
+    else:
+        print("ERROR: Incorrect value '" + geometry_string +
+              "' for geometry (must be in the form: x:z+width+height)")
+        usage()
+        sys.exit(2)
+elif region_string is not None:
+    # parts = region_string.split(" ")
+    axis_info = region_string.split(",")
+    if len(axis_info) == 2:
+        # xmin_string, xmax_string, zmin_string, zmax_string = parts
+        x_bounds, z_bounds = axis_info
+        xmin_string, xmax_string = x_bounds.split(":")
+        zmin_string, zmax_string = z_bounds.split(":")
+        nonchunky_xmin = int(xmin_string)
+        nonchunky_xmax = int(xmax_string)
+        nonchunky_zmin = int(zmin_string)
+        nonchunky_zmax = int(zmax_string)
+        print("region:")
+        print("  xmin:" + str(nonchunky_xmin))
+        print("  xmax:" + str(nonchunky_xmax))
+        print("  zmin:" + str(nonchunky_zmin))
+        print("  zmax:" + str(nonchunky_zmax))
+    else:
+        print("ERROR: Incorrect value '" + region_string +
+              "' for region (must be in the form: xmin:xmax,zmin:zmax)")
+        usage()
+        sys.exit(2)
+
+sector_xmin = nonchunky_xmin / 16
+sector_xmax = nonchunky_xmax / 16
+sector_zmin = nonchunky_zmin / 16
+sector_zmax = nonchunky_zmax / 16
+
 if path is None:
-    print("Please select world path (eg. -i ../worlds/yourworld) (or use --help)")
+    print("Please select world path (-i ../worlds/yourworld) (or use --help)")
     sys.exit(1)
 
 if path[-1:] != "/" and path[-1:] != "\\":
     path = path + "/"
 
 # Load color information for the blocks.
 colors = {}
+colors_path = "colors.txt"
+
+profile_path = None
+if 'USERPROFILE' in os.environ:
+    profile_path = os.environ['USERPROFILE']
+elif 'HOME' in os.environ:
+    profile_path = os.environ['HOME']
+
+mt_path = None
+mt_util_path = None
+abs_colors_path = None
+
+if not os.path.isfile(colors_path):
+    colors_path = os.path.join(os.path.dirname(os.path.abspath(__file__)),
+                               "colors.txt")
+
+if not os.path.isfile(colors_path):
+    if profile_path is not None:
+        try_path = os.path.join(profile_path, "minetest")
```

Comment:
I thought using the profile_path would be ok since it is now just for fault tolerance (for one step in finding try_path), while a colors.txt in the same folder takes priority, in which case profile_path will not be used. Let me know if you still think it should be changed.